### PR TITLE
Add metrics to service factory

### PR DESCRIPTION
**Pull request by carabasdaniel**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,26 +1,44 @@
...
+// replace github.com/aserto-dev/aserto-grpc => ../aserto-grpc
+
 go 1.19
 
 require (
+	github.com/aserto-dev/aserto-grpc v0.2.2
 	github.com/aserto-dev/certs v0.0.2
 	github.com/aserto-dev/go-http-metrics v0.10.1-20221024-1
-	github.com/grpc-ecosystem/grpc-gateway/v2 v2.7.3
+	github.com/grpc-ecosystem/go-grpc-middleware/providers/prometheus v1.0.0-rc.0
```
Comment (gertd) [inline go.mod:11]: Why are we using rc0 if v1.4.0 is available?

File: go.mod [hunk 2]
```diff
@@ -1,26 +1,44 @@
...
+	github.com/grpc-ecosystem/grpc-gateway/v2 v2.15.2
```
Comment (gertd) [inline go.mod]: Please update to v2.16.0

File: go.mod [hunk 3]
```diff
@@ -1,26 +1,44 @@
...
 	github.com/pkg/errors v0.9.1
+	github.com/prometheus/client_golang v1.14.0
```
Comment (gertd) [inline go.mod]: Can we use the latest released v1.16.0?

File: go.mod [hunk 4]
```diff
@@ -1,26 +1,44 @@
...
 	github.com/rs/cors v1.9.0
 	github.com/rs/zerolog v1.29.1
-	golang.org/x/sync v0.0.0-20190423024810-112230192c58
-	google.golang.org/grpc v1.49.0
-	google.golang.org/protobuf v1.27.1
+	go.opencensus.io v0.24.0
+	golang.org/x/sync v0.1.0
+	google.golang.org/grpc v1.54.0
```
Comment (gertd) [inline go.mod]: We use v1.56.2

File: go.mod [hunk 5]
```diff
@@ -1,26 +1,44 @@
...
+	google.golang.org/protobuf v1.30.0
```
Comment (gertd) [inline go.mod]: We use v1.13.1

File: service_factory.go [hunk 6]
```diff
@@ -48,14 +70,34 @@ func (f *ServiceFactory) CreateService(config *API, opts []grpc.ServerOption, re
...
+		reg := prometheus.NewRegistry()
+
+		grpcm.InitializeMetrics(grpcServer)
+		reg.MustRegister(collectors.NewGoCollector())
+		reg.MustRegister(collectors.NewBuildInfoCollector())
+		reg.MustRegister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{ReportErrors: true}))
+		reg.MustRegister(grpcm)
+		mux.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{}))
+
+		// TODO: Fix zpages
```
Comment (gertd) [inline service_factory.go]: We had decided to remove zpages, so we can remove all references

**Review by gertd — CHANGES_REQUESTED** [review]
Please update dependencies and remove zpages reference, rest look good

File: go.mod (hunk 1 above)
Comment (carabasdaniel) [reply go.mod:11]: v1.0.0-rc.0 is latest release for prometheus provider

**Review by gertd — APPROVED** [review]

**Pull request merged**